quantum/process_keycode/process_dynamic_macro.c

```diff
@@ -219,10 +221,8 @@ bool process_dynamic_macro(uint16_t keycode, keyrecord_t *record) {
             case DYN_REC_START1:
             case DYN_REC_START2:
             case DYN_REC_STOP:
-                /* Stop the macro recording. */
-                if (record->event.pressed ^ (keycode != DYN_REC_STOP)) { /* Ignore the initial release
-                                                                          * just after the recording
-                                                                          * starts for DYN_REC_STOP. */
+                /* Stop the macro recording on key release. Either way, don't record it. */
+                if (!record->event.pressed) {
```

Comment:
Impossible to start recording with the stop key, so this if statement could be simplified. I suspect this also makes holding TT and hitting stop work as expected.